libraries/csharp-net22-runtime/context/Pipe/Program.cs

```diff
@@ -0,0 +1,53 @@
+﻿using System;
+
+namespace Pipe
+{
+    class Program
+    {
+        static int Main(string[] args)
+        {
+            if (args.Length == 0)
+            {
+                throw new Exception(
+                    "no algorithm directory argument found. Please provide the path to a valid C# algorithm.");
+            }
+            string sysPath = args[0];
+            Config config = new Config(sysPath);
+            Module algoModule;
+            try
+            {
+                algoModule = new Module(config);
+            }
+            catch (Exception e)
+            {
+                ExceptionResponse response = new ExceptionResponse(e);
+                Write.WriteJsonToPipe(response);
+                Console.Out.WriteLine("PIPE_INIT_FAILED");
+                return -1;
+            }
+            Console.Out.WriteLine("PIPE_INIT_COMPLETE");
+            
+            string readLine;
+            while ((readLine = Console.In.ReadLine()) != null && readLine != "")
+            {
+                Request request = new Request(readLine);
+                object response = null;
+                try
+                {
+                    object result = algoModule.AttemptExecute(request);
+                     response = new Response(result, "json");
+                }
+                catch (Exception e)
+                {
+                    response = new ExceptionResponse(e);
+                }
+                finally
+                {
+                    Write.WriteJsonToPipe(response);

```

Comment:
should validate that response is not null before writing it to the pipe.  if it is null then fill in a default error message